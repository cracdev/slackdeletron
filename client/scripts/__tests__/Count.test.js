import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import Count from '../Components/Count';

const MOCK_DATA = [{ size: 300 }, { size: 200 }, { size: 500 }];

describe('<Checkbox />', () => {
  it('renders correctly', () => {
    const count = renderer.create(<Count />).toJSON();
    expect(count).toMatchSnapshot();
  });

  it('displays the correct number of files', () => {
    const count = shallow(<Count data={MOCK_DATA} total={MOCK_DATA.length} />);
    expect(count.find('.purple').text()).toBe('3 files');
  });

  it('displays the correct number of files and teamname', () => {
    const count = shallow(<Count data={MOCK_DATA} total={MOCK_DATA.length} />);
    expect(
      count
        .find('.Count__Text')
        .at(0)
        .text()
    ).toBe('There are 3 files you can delete from your workspace.');
  });

  it('displays the right file size', () => {
    const count = shallow(<Count data={MOCK_DATA} total={MOCK_DATA.length} />);
    expect(
      count
        .find('.Count__Text')
        .at(1)
        .text()
    ).toBe(`Here are ${MOCK_DATA.length} worth 1000 Bytes`);
  });
});
