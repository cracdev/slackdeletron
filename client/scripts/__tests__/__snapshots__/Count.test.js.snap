// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Checkbox /> renders correctly 1`] = `
<div
  className="Count"
>
  <p
    className="Count__Text"
  >
    There is
     
    <span
      className="purple"
    >
       
      file
    </span>
     
    you can delete from your workspace.
  </p>
  <p
    className="Count__Text"
  >
    Here is
     
    0
     worth
     
    <span
      className="red"
    >
      0 Bytes
    </span>
  </p>
</div>
`;
